cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -996,6 +998,22 @@ public void close(QueryHandle handle) throws ExploreException, HandleNotFoundExc
     return result;
   }
 
+  @Override
+  public int getActiveQueryCount(Id.Namespace namespace) throws ExploreException {
+    startAndWait();
+
+    int count = 0;
+    for (Map.Entry<QueryHandle, OperationInfo> entry : activeHandleCache.asMap().entrySet()) {
+      if (entry.getValue().getNamespace().equals(getHiveDatabase(namespace.getId()))) {
```

Comment:
`getHiveDatabase(namespace.getId())` does not change for a method call, it can move out of the loop, right?